File: swapi-server/src/main/java/de/lkaktus/entities/SpeciesDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record SpeciesDTO(
        @JsonProperty("average_height") String averageHeight,
        @JsonProperty("average_lifespan") String averageLifespan,
        String classification,
        String created,
        String designation,
        String edited,
        @JsonProperty("eye_colors") String eyeColors,
        @JsonProperty("hair_colors") String hairColors,
        String homeworld,
        SearchItemDTO homeworldItem,
        String language,
        String name,
        @JsonProperty("skin_colors") String skinColors,
        String url,
        String [] people,
        SearchItemDTO [] characterItems,
        String [] films,
        SearchItemDTO [] filmItems
) {

}

File: swapi-angular/src/app/api/models/SpeciesDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type SpeciesDTO = {
    average_height?: string;
    average_lifespan?: string;
    eye_colors?: string;
    hair_colors?: string;
    skin_colors?: string;
    classification?: string;
    created?: string;
    designation?: string;
    edited?: string;
    homeworld?: string;
    homeworldItem?: SearchItemDTO;
    language?: string;
    name?: string;
    url?: string;
    people?: Array<string>;
    characterItems?: Array<SearchItemDTO>;
    films?: Array<string>;
    filmItems?: Array<SearchItemDTO>;
};

